### PR TITLE
OWASP dependency checks

### PR DESCRIPTION
**Pull request by jaahon**
#### Summary

Add OWASP dependency checks to service and message service.

Update some dependencies with vulnerabilities.

Suppress warnings for false positives and dependencies we can't do anything about.

### CODE REVIEW

File: message-service/build.gradle.kts [hunk 1]
```diff
@@ -202,4 +203,10 @@ tasks {
         classpath = ktlint
         args = listOf("applyToIDEAProject", "-y")
     }
+
+    dependencyCheck {
+        failBuildOnCVSS = 0.0f
+        analyzers.assemblyEnabled = false
```
Comment (mikkopiu) [inline message-service/build.gradle.kts:209]: In my experience, it's best to disable **all** the unused analyzers as the plugin will spend some unnecessary time initializing and attempting to find those non-existent dependencies:
```suggestion
        analyzers(delegateClosureOf<org.owasp.dependencycheck.gradle.extension.AnalyzerExtension> {
            assemblyEnabled = false
            nodeAuditEnabled = false
            nodeEnabled = false
            nuspecEnabled = false
        })
```

**Review by mikkopiu — CHANGES_REQUESTED** [review]
Looking good overall, a few nits + need to investigate why the job is still so slow even with a fresh NVD cache.

Also, I think we should store the dependency-check report as an artifact so it's a bit easier to find & read than just the console output.

File: .circleci/config.yml [hunk 2]
```diff
@@ -749,6 +750,32 @@ jobs:
...
+          cve_data_directory: ~/repo/.gradle-user-home/dependency-check-data
+          cache_key: gradle-v1-cache-key
+      - run:
+          name: Run OWASP Dependency-Check Analyzer
+          working_directory: *workspace_service
+          command: ./gradlew dependencyCheckAnalyze
+      - run:
+          name: OWASP Dependency-Check cache cleanup
+          working_directory: *workspace_service
+          command: ./gradlew dependencyCheckPurge
```
[the remark this answers is not in the document]
Comment (jaahon) [reply .circleci/config.yml:776]: The slack orb does it in its jobs so I just copied it without thinking 😄

File: .circleci/config.yml [hunk 3]
```diff
@@ -749,6 +750,32 @@ jobs:
...
+      - *restore_repo
+      - *restore_service_gradle
+      - owasp/generate_cache_keys:
+          cache_key: gradle-v1-cache-key
+      - owasp/restore_owasp_cache:
+          cache_key: gradle-v1-cache-key
+      - run:
+          name: Update OWASP Dependency-Check Database
+          working_directory: *workspace_service
+          command: ./gradlew dependencyCheckUpdate
```
[the remark this answers is not in the document]
Comment (mikkopiu) [reply .circleci/config.yml:765]: Hmm, maybe it was just a incomplete cache or something, 'cause now that I'm re-running the same job the update step takes on ~15s (reasonable, not great but don't think it'll ever be)

**Review by mikkopiu — APPROVED** [review]
Great stuff. I also verified from the dependency-check report that it is indeed scanning the service-lib dependencies as desired.

**Pull request merged**